rpcserver.go

```diff
@@ -7785,6 +7790,7 @@ func (r *rpcServer) AddInvoice(ctx context.Context,
 		return nil, fmt.Errorf("invoice request must be specified")
 	}
 	iReq := req.InvoiceRequest
+	existingQuotes := iReq.RouteHints != nil
```

Comment:
Hmm, perhaps we should actually make validation stricter here and _disallow_ adding extra route hints? Otherwise a user can add hop hints that affect our reachability, or encodes the wrong details, etc.